src/wallet/wallet.cpp

```diff
@@ -1062,8 +1064,23 @@ bool CWallet::AddToWalletIfInvolvingMe(const CTransactionRef& ptx, CWalletTx::Co
 
             // loop though all outputs
             for (const CTxOut& txout: tx.vout) {
-                for (const auto& spk_man_pair : m_spk_managers) {
-                    spk_man_pair.second->MarkUnusedAddresses(txout.scriptPubKey);
+                const auto& spk_man = GetScriptPubKeyMan(txout.scriptPubKey);
```

Comment:
@achow101 I think it's technically possible to have two descriptors producing same scriptPubKey. Wdyt?

Reply:
Ah, that is indeed possible. In that case, we should keep this loop.

Reply:
Returned the cycle and removed `GetScriptPubKeyMan` function to avoid such mistakes in the future